kotlin-sqlobject/src/main/kotlin/org/jdbi/v3/sqlobject/kotlin/Extensions.kt

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.sqlobject.kotlin
+
+import org.jdbi.v3.core.Handle
+import org.jdbi.v3.core.Jdbi
+import kotlin.reflect.KClass
+
+fun <T : Any> Jdbi.onDemand(sqlObjectType: KClass<T>): T {
+    return this.onDemand(sqlObjectType.java)
+}
+
+// inline fun <reified T: Any> DBI.onDemand(): T {
```

Comment:
Actually I think I prefer the commented out version with type parameter:

`db.jdbi.onDemand<ThingDao>()` vs `db.jdbi.onDemand(ThingDao::class))`